tests/src/cosmwasm.spec.ts

```diff
@@ -315,3 +316,57 @@ test.serial("properly rollback first submessage if second fails", async (t) => {
   const gotNoFunds = await osmoClient.sign.getBalance(emptyAddr, osmosis.denomFee);
   t.is(gotNoFunds.amount, "0");
 });
+
+test.serial("query remote chain", async (t) => {
```

Comment:
Nice.
We also need to test the callback logic. But already this is a lot of good stuff